tests/ignite/engine/test_create_supervised.py

```diff
@@ -106,7 +106,7 @@ def _():
         assert pytest.approx(trainer.state.output[-1], abs=1e-5) == loss[0]
         accumulation[0] = loss[0] = 0.0
 
-    if model_device == trainer_device or ((model_device == "cpu") ^ (trainer_device == "cpu")):
+    if model_device == trainer_device or ((model_device in ["cpu", None]) and (trainer_device in ["cpu", None])):
```

Comment:
I changed this because in the case `trainer_device` and `model_device` were `cuda` and `cpu` or vice versa, we would not catch device mismatch error.